src/mongo/shell.ts

```diff
@@ -5,23 +5,30 @@
 import * as cp from 'child_process';
 import * as os from 'os';
 import { IDisposable, toDisposable } from '../utils/vscodeUtils';
-import { EventEmitter, window } from 'vscode';
+import { EventEmitter } from 'vscode';
 import { ext } from '../extensionVariables';
 
+type CommandResult = {
```

Comment:
This is the crux of the fix...  Previously all errors were processed just like the actual data.  I've fixed all error code so that the processing can correctly tell what are errors and what are results.